appservice/src/createAppService/SiteNameStep.ts

```diff
@@ -59,15 +59,16 @@ export class SiteNameStep extends AzureNameStep<SiteNameStepWizardContext> {
             prompt = vscode.l10n.t('Enter a globally unique name for the new web app.');
         }
 
+        const agentMetadata = this._siteFor === "functionApp" ?
+            { parameterDisplayTitle: vscode.l10n.t('Function App Name'), parameterDisplayDescription: vscode.l10n.t('The name of the new function app.') } :
+            { parameterDisplayTitle: vscode.l10n.t('Site Name'), parameterDisplayDescription: vscode.l10n.t('The name of the app service site.'), };
+
         const options: AgentInputBoxOptions = {
             prompt,
             placeHolder,
             validateInput: (name: string): string | undefined => this.validateSiteName(name),
             asyncValidationTask: async (name: string): Promise<string | undefined> => await this.asyncValidateSiteName(client, name),
-            agentMetadata: this._inputBoxOptionsAgentMetadata ?? {
-                parameterDisplayTitle: "Site Name",
-                parameterDisplayDescription: "The name of the app service site.",
-            }
+            agentMetadata: agentMetadata
```

Comment:
I KNOW YOU WON'T CHANGE IT, BUT

nit:

```suggestion
            agentMetadata
```